File: php/src/Model/SupervisorParkedTask.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A parked task as the board shows it — enough to decide on, not the whole task.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class SupervisorParkedTask
{
    public function __construct(
        public readonly string $id,
        /** @var list<string> */
        public readonly array $tags,
        public readonly ?float $priority,
        public readonly ?int $createdAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            tags: \array_values(\array_map(strval(...), $data['tags'] ?? [])),
            priority: isset($data['priority']) ? (float) $data['priority'] : null,
            createdAt: isset($data['createdAt']) ? (int) $data['createdAt'] : null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/SupervisorParkedTask.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * A parked task as the board shows it — enough to decide on, not the whole task.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class SupervisorParkedTask {
    private String id;
    private List<String> tags;
    private Double priority;
    private Long createdAt;

    public String getId() { return id; }
    public List<String> getTags() { return tags; }
    public Double getPriority() { return priority; }
    public Long getCreatedAt() { return createdAt; }
}
